src/Support/ResolveInfoFieldsAndArguments.php

```diff
@@ -132,15 +143,67 @@ private function foldSelectionSet(SelectionSetNode $selectionSet, int $descend):
         return $fields;
     }
 
-    private function getValue(ArgumentNode $argumentNode)
+    /**
+     * @param ValueNode $value
+     * @return mixed
+     * @throws \Exception
+     */
+    private function getValue(ValueNode $value)
     {
-        $value = $argumentNode->value;
         if ($value instanceof VariableNode) {
             $variableName = $value->name->value;
 
             return $this->info->variableValues[$variableName] ?? null;
         }
+        // Scalar Types
+        if ($value instanceof IntValueNode || $value instanceof FloatValueNode || $value instanceof StringValueNode || $value instanceof BooleanValueNode || $value instanceof EnumValueNode) {
+            return $value->value;
+        }
+        // null Type
+        if ($value instanceof NullValueNode) {
+            return;
+        }
+        // object Type
+        if ($value instanceof ObjectValueNode) {
+            return $this->getInputObjectValue($value);
+        }
+        // list object Type
+        if ($value instanceof ListValueNode) {
+            return $this->getInputListObjectValue($value);
+        }
+
+        throw new \Exception('Failed to resolve unknown ValueNode type');
```

Comment:
If you don't add a specific exception, please use at least `RuntimeException` (and import it)

Reply:
How can this be triggered with the current code?

Would like to see a test for this.